teos/src/config.rs

```diff
@@ -321,46 +359,74 @@ mod tests {
         assert_eq!(config, config_clone);
     }
 
-    #[test]
-    fn test_config_default_not_verify() {
+    #[tokio::test]
+    async fn test_config_default_not_verify() {
         // Tests that the default configuration does not pass verification checks. This is on purpose so some fields are
         // required to be updated by the user.
         let mut config = Config::default();
-        assert!(matches!(config.verify(), Err(ConfigError { .. })));
+        assert!(
+            matches!(config.verify().await, Err(ConfigError(e)) if e.contains("btc_rpc_user must be set"))
+        );
     }
 
-    #[test]
-    fn test_config_default_verify_overwrite_required() {
+    #[tokio::test]
+    async fn test_config_default_verify_overwrite_required() {
+        let (_mock, ip, port) = get_bitcoind_mock();
         // Tests that setting a some btc_rpc_user and btc_rpc_password results in a Config object that verifies
         let mut config = Config {
             btc_rpc_user: "user".to_owned(),
             btc_rpc_password: "password".to_owned(),
+            btc_rpc_connect: ip,
```

Comment:
This is not necessary, is it?

The idea behind this test is that just setting the minimal require stuff (`btc_rpc_user` and `btc_rpc_password` which are set empty by default on purpose) the resulting config does verify.

Reply:
Now that verify does the whole verification pipeline (checks our network against bitcoind's network), we need to pass the ip and port to a bitcoind instance for this not to fail.

Reply:
We won't if this is postponed to `BitcoindClient::new` right?

Reply:
> We won't if this is postponed to BitcoindClient::new right?

Yes